bin/restore_gradle_dependency_cache

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash -eu
+
+# The key is shared with bin/restore_gradle_dependency_cache
+GRADLE_DEPENDENCY_CACHE_KEY="GRADLE_DEPENDENCY_CACHE"
```

Comment:
While looking at `mobile-secrets` to see the changes that added the env var mentioned in this PR, I noticed that this value was originally in the `env` file, but then got removed. Why was that?

Has that got to do with the following comment in the `save_...` file?

> For now we are using a single key - we might expand on this later by using dependency catalog version

Reply:
This is a global key and I thought it made sense to keep it in one place - save & restore scripts. There is some possibility to calculate the key in the future (using the catalog version for example), so having it directly in the script makes that easier.

I'd have liked to define everything in this plugin, but for consistency it made more sense to do it in `.mobile-secrets` or the AMI configuration. If we are not against doing it as a plugin hook, I'd be happy to move everything here. I am also happy to move this key back to `.mobile-secrets` env files if you'd prefer. I don't have any strong opinions, just doing what makes the most sense in the current structure we have.

Reply:
> This is a global key and I thought it made sense to keep it in one place - save & restore scripts. There is some possibility to calculate the key in the future (using the catalog version for example), so having it directly in the script makes that easier.

👍